cmd/simulacra/simulacra.go

```diff
@@ -0,0 +1,183 @@
+	}
+	return builder.String()
+}
+
+// envVarMapToPowershellPrefix converts a map of env variable name to value into a string
+// suitable for prepending onto a powershell command as a way to set those variables.
+// Example output: "$env:VAR1='foo'\n$env:VAR2='bar'\n"
+func envVarMapToPowershellPrefix(env map[string]string) string {
+	var builder strings.Builder
+	for key, value := range env {
+		fmt.Fprintf(&builder, "$env:%s='%s'\n", key, value)
+	}
+	return builder.String()
+}
+
+func runScriptRemotely(ctx context.Context, logger *log.Logger, vm *gce.VM, scriptContents string, env map[string]string) (_ gce.CommandOutput, err error) {
+	if gce.IsWindows(vm.Platform) {
+		// Use a UUID for the script name in case RunScriptRemotely is being
+		// called concurrently on the same VM.
+		scriptPath := "C:\\" + uuid.NewString() + ".ps1"
+		if err := gce.UploadContent(ctx, logger, vm, strings.NewReader(scriptContents), scriptPath); err != nil {
+			return gce.CommandOutput{}, err
+		}
+		// powershell -File seems to drop certain kinds of errors:
+		// https://stackoverflow.com/a/15779295
+		// In testing, adding $ErrorActionPreference = 'Stop' to the start of each
+		// script seems to work around this completely.
+		return gce.RunRemotely(ctx, logger, vm, "", envVarMapToPowershellPrefix(env)+"powershell -File "+scriptPath)
+	}
+	scriptPath := uuid.NewString() + ".sh"
+	// Write the script contents to <UUID>.sh, then tell bash to execute it with -x
+	// to print each line as it runs.
+	// Use a UUID for the script name in case RunScriptRemotely is being called
+	// concurrently on the same VM.
+	return gce.RunRemotely(ctx, logger, vm, scriptContents, "cat - > "+scriptPath+" && sudo "+envVarMapToBashPrefix(env)+"bash -x "+scriptPath)
+}
+
+// installApps reads an Ops Agent config file and then identifies all the third party apps that need to be installed.
+// The function identifies third party apps to install by checking if any of the receiver types have a
+// corresponding install script in the third_party_apps_data directory.
+// If there is a corresponding install script, then that install script is run on the vm.
+func installApps(ctx context.Context, vm *gce.VM, logger *log.Logger, configFilePath string, installPath string) error {
+	config, err := confgenerator.MergeConfFiles(ctx, configFilePath, apps.BuiltInConfStructs)
+	if err != nil {
+		return err
+	}
+
+	folder, err := distroFolder(vm.Platform)
+
+	if err != nil {
+		return err
+	}
+
+	receivers := getAllReceivers(config)
+
+	for _, app := range receivers {
+		if scriptContent, err := os.ReadFile(path.Join(installPath, "applications", app, folder, "install")); err == nil {
+			logger.Printf("Installing %s to VM", app)
+			if _, err := runScriptRemotely(ctx, logger, vm, string(scriptContent), make(map[string]string)); err != nil {
+				return err
+			} else {
+				logger.Printf("Done Installing %s", app)
+			}
+		}
+	}
+	return nil
+}
+
+func main() {
```

Comment:
Can be done in another PR, but we will want simulacra to be configured using YAML. That way the configs can be shared and we can have automation in some places (like diagnostic script creating a config for the tool).

Also, when we go into some of the more tricky cases, I imagine we will want yaml anyway like specifying VM details like name, ram, disk, vcpus etc.